packages/react-auth/src/ComponentAuthProvider/index.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  FC,
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useState,
+  useMemo,
+  useEffect,
+} from "react";
+
+import AuthNClient from "@src/AuthNClient";
+
+import { APIResponse, AuthNConfig, Token, AuthUser } from "@src/types";
+
+export interface SessionData {
+  user?: AuthUser;
+}
+
+export interface ComponentAuthContextType {
+  authenticated: boolean;
+  loading: boolean;
+  error: APIResponse | undefined;
+  user?: AuthUser;
+  login: () => void;
+  logout: () => void;
+  config: AuthNConfig;
+  setFlowComplete: (data: any) => void;
+}
+
+export interface ComponentAuthProps {
+  config: AuthNConfig;
+  children: ReactNode;
+}
+
+const SESSION_NAME = "pangea-session";
+
+const PangeaContext = createContext<ComponentAuthContextType>(
+  {} as ComponentAuthContextType
+);
+
+const ComponentAuthProvider: FC<ComponentAuthProps> = ({
+  config,
+  children,
+}) => {
+  const [authenticated, setAuthenticated] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<APIResponse>();
+  const [user, setUser] = useState<AuthUser>();
+
+  const auth = new AuthNClient(config);
+
+  // load data from local storage, and params from URL
+  useEffect(() => {
+    const sessionData = getSessionData();
+
+    if (sessionData?.user?.active_token?.token) {
+      validate(sessionData);
+    } else {
+      setLoading(false);
+    }
+
+    // eslint-disable-next-line
+  }, []);
+
+  // update session data on step
+  useEffect(() => {
+    const sessionData = getSessionData();
+    sessionData.user = user;
+    saveSessionData(sessionData);
+
+    // eslint-disable-next-line
+  }, [user]);
+
+  const getSessionData = (): SessionData => {
+    const storedData = localStorage.getItem(SESSION_NAME);
+
+    if (storedData) {
+      const session: SessionData = JSON.parse(storedData);
+      return session;
+    }
+
+    return {};
+  };
+
+  const saveSessionData = (data: SessionData) => {
+    const dataString = JSON.stringify(data);
+    localStorage.setItem(SESSION_NAME, dataString);
+  };
+
+  const validate = async (data: SessionData) => {
```

Comment:
shared?

Reply:
The `validate` methods in the providers are different in how they handle the response, so I left them separate. Both call the same `client.validate()` method though.